insights/parsers/engine_config.py

```diff
@@ -0,0 +1,83 @@
+from collections import namedtuple
+from operator import itemgetter
+from .. import Parser, parser, get_active_lines
+
+
+@parser("engine-config_all")
+class EngineConfigAll(Parser):
+    """
+    Sample output::
+
+    AffinityRulesEnforcementManagerEnabled: true version: general
+    AffinityRulesEnforcementManagerRegularInterval: 1 version: general
+    MaxNumOfVmSockets: 16 version: 4.1
+    MaxRerunVmOnVdsCount: 3 version: general
+    MaxSchedulerWeight: 1000 version: general
+    MaxStorageVdsDelayCheckSec: 5 version: general
+    MaxStorageVdsTimeoutCheckSec: 30 version: general
+    MaxVdsMemOverCommit: 200 version: general
+    MaxVdsMemOverCommitForServers: 150 version: general
+    MaxVdsNameLength: 255 version: general
+    MaxVmNameLength: 64 version: general
+    MaxVmNameLengthSysprep: 15 version: general
+
+    Examples::
+    >>> from insights.parsers.engine_config import EngineConfigAll
+    >>> from insights.tests import context_wrap
+    >>> shared = {EngineConfigAll: EngineConfigAll(context_wrap(OUTPUT))}
+    >>> output = shared[EngineConfigAll]
+    >>> output.['MaxStorageVdsTimeoutCheckSec']
+    ['30']
+    >>> output.head('MaxStorageVdsTimeoutCheckSec') == '30'
+    '30'
+    >>> output.get_version('MaxStorageVdsTimeoutCheckSec')
+    ['general']
+    >>> output['ClusterRequiredRngSourcesDefault']
+    []
+    >>> output.get_version('ClusterRequiredRngSourcesDefault')
+    ['3.6', '4.0', '4.1']
+    >>> output.last('HotUnplugCpuSupported')
+    '{"x86":"true","ppc":"true"}'
+    """
+    KeyValue = namedtuple('KeyValue',
+                          ['keyword', 'value', 'version', 'kw_lower'])
+
+    def parse_content(self, content):
+        self.fields = []
+        for line in get_active_lines(content):
+            try:
+                key, val, ver = itemgetter(0, 1, -1)(line.split(" "))
+                self.fields.append(self.KeyValue(key.strip(':'), val, ver, key.strip(':').lower()))  # noqa
+            except:
+                pass
+        self.keywords = set([kw.kw_lower for kw in self.fields])
```

Comment:
Better explain it clearly you have converted all keywords to lower cases in the document.

Reply:
Sure